### ISSUE TITLE
Cannot use Java 11 HttpClient instrumentation in Kotlin application

### ISSUE THREAD

**Issue by gmariottiAtWolt** [micrometer-metrics/micrometer#3577]
**Describe the bug**
This is a follow-up to [this message](https://github.com/micrometer-metrics/micrometer/issues/2511#issuecomment-1359381235) in the HttpClient Micrometer instrumentation. It seems that the multi-release jar can't be consumed in a Kotlin+Gradle application using Java 11+.

**Environment**

 - Micrometer version: 1.10.2
 - ~Micrometer registry [e.g. prometheus]~
 - ~OS: [e.g. macOS]~
 - Java version: 17

**To Reproduce**
How to reproduce the bug:

Create a Gradle project with

```kotlin
// build.gradle.kts
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile

plugins {
    kotlin("jvm") version "1.7.20"
}

repositories {
    mavenCentral()
}

java {
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}

tasks.withType<KotlinCompile> {
    kotlinOptions.jvmTarget = "17"
}

dependencies {
    implementation("io.micrometer:micrometer-core:1.10.2")
}
```

```kotlin
// src/main/kotlin/main.kt
import io.micrometer.core.instrument.binder.jdk.MicrometerHttpClient
import io.micrometer.core.instrument.simple.SimpleMeterRegistry
import java.net.http.HttpClient

fun main() {
    val client = MicrometerHttpClient.instrumentationBuilder(HttpClient.newHttpClient(), SimpleMeterRegistry()).build()
}
```

then run `./gradle compileKotlin`

**Expected behavior**
The Gradle task completes without any exception (making it possible to import `MicrometerHttpClient`)

**Additional context**
Add any other context about the problem here, e.g. related issues.

**Comment by LWogan** [comment]
I've come across the same issue. Is there any known workarounds?

**Comment by marcingrzejszczak** [comment]
Is this a problem only in the IDE or is there a problem during the build?

**Comment by gmariottiAtWolt** [comment]
Hi @marcingrzejszczak, the problem is the build. I just tried with the updated `build.gradle.kts` below, and the project fails to compile with

```
> Task :compileKotlin FAILED
e: file:///Users/gmariotti/temp-project/src/main/kotlin/main.kt:1:45 Unresolved reference: jdk
e: file:///Users/gmariotti/temp-project/src/main/kotlin/main.kt:6:18 Unresolved reference: MicrometerHttpClient
```

```kotlin
import org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_17
import org.jetbrains.kotlin.gradle.dsl.KotlinVersion.KOTLIN_1_9
import org.jetbrains.kotlin.gradle.tasks.KotlinJvmCompile

plugins {
    kotlin("jvm") version "1.9.22"
}

repositories {
    mavenCentral()
}

java {
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}

tasks.withType<KotlinJvmCompile> {
    compilerOptions {
        jvmTarget = JVM_17
        languageVersion = KOTLIN_1_9
        apiVersion = KOTLIN_1_9
    }
}

dependencies {
    implementation("io.micrometer:micrometer-core:1.12.1")
}
```

**Comment by jonatan-ivanov** [comment]
I'm not 100% that this helps but we fixed things around kotlin and okhttp versions: https://github.com/micrometer-metrics/micrometer/pull/4580

Can you try with the latest snapshots?

**Comment by monosoul** [comment]
@jonatan-ivanov I don't think okhttp version had anything to do with the original issue here, I think it has more to do with Gradle itself. I think in addition to having a multi-release JAR you can simply publish a separate JAR that will include all classes without multi-release magic. Here's a PR to do that: https://github.com/micrometer-metrics/micrometer/pull/4728

**Comment by shakuzen** [comment]
Thanks to the work by @monosoul, we will have a new module micrometer-java11 that contains the Java 11 HttpClient instrumentation, which should avoid the multi-release jar issues related to using this in Kotlin. I tried out the new module locally with a sample Kotlin app, but if other users could [try it out](https://github.com/micrometer-metrics/micrometer/?tab=readme-ov-file#snapshot-builds) with 1.13.0-SNAPSHOT, that would be great. The new module will also be included in the upcoming 1.13.0-M2 milestone preview release.

**Comment by monosoul** [comment]
@shakuzen just tried it out, works fine. Thanks!
